Quick starts example 'Using HTTP with Reactive Messaging' is not working

### Describe the bug

Following ['Using HTTP with Reactive Messaging'](https://quarkus.io/guides/reactive-messaging-http.html) 
I cloned the repo and started the app. However, after clicking add price on [the HTML page](http://localhost:8080/index.html)
I get an error in log 

```java
2021-09-20 10:25:03,473 ERROR [io.qua.rea.htt.run.ReactiveHttpHandlerBean] (vert.x-eventloop-thread-13) Failed to process message.: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:707)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

### Expected behavior

the app should work as described in the guide

### Actual behavior

_No response_

### How to Reproduce?

Follow [the guide](https://quarkus.io/guides/reactive-messaging-http.html)  by cloning the repo, checkout 2.2.3.Final tag and watch the console

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/docs/src/main/asciidoc/reactive-messaging-http.adoc
+++ b/docs/src/main/asciidoc/reactive-messaging-http.adoc
@@ -154,9 +154,12 @@ The `channel-name` segment must match the value set in the `@Incoming` and `@Out
 ----
 mp.messaging.outgoing.outgoing-costs.connector=quarkus-http
 
-# here we are using a URL pointing to a test endpoint
+# here we are using a URL pointing to an endpoint
 # you can use e.g. an environment variable to change it
 mp.messaging.outgoing.outgoing-costs.url=http://localhost:${quarkus.http.port}/cost-collector
+
+# we need to use a different port for tests:
+%test.mp.messaging.outgoing.outgoing-costs.url=http://localhost:${quarkus.http.test-port}/cost-collector
 
 # POST is the default method. Another possibility is PUT
 mp.messaging.outgoing.outgoing-costs.method=POST

